Remove default allocator assignment in clap

I am working on a project which uses clap in UEFI and having the default allocator in the structs is causing this error despite me overriding the allocator.
```
/home/ross/zig/lib/zig/std/os.zig:283:25: error: root struct of file 'os.uefi' has no member named 'getErrno'
pub const errno = system.getErrno;
                  ~~~~~~^~~~~~~~~
/home/ross/zig/lib/zig/std/os/uefi.zig:1:1: note: struct declared here
const std = @import("../std.zig");
^~~~~
referenced by:
    munmap: /home/ross/zig/lib/zig/std/os.zig:4539:13
    free: /home/ross/zig/lib/zig/std/heap/PageAllocator.zig:108:11
    vtable: /home/ross/zig/lib/zig/std/heap/PageAllocator.zig:13:13
    page_allocator: /home/ross/zig/lib/zig/std/heap.zig:241:33
    ParseOptions: /home/ross/.cache/zig/p/122066d86920926a4a674adaaa10f158dc4961c2a47e588e605765455d74ca72c2ad/clap.zig:649:36
    ParseOptions: /home/ross/.cache/zig/p/122066d86920926a4a674adaaa10f158dc4961c2a47e588e605765455d74ca72c2ad/clap.zig:643:26
    parseEx: /home/ross/.cache/zig/p/122066d86920926a4a674adaaa10f158dc4961c2a47e588e605765455d74ca72c2ad/clap.zig:728:10
    parse__anon_2995: /home/ross/.cache/zig/p/122066d86920926a4a674adaaa10f158dc4961c2a47e588e605765455d74ca72c2ad/clap.zig:666:24
    main: src/ziggybox.zig:20:25
    EfiMain: /home/ross/zig/lib/zig/std/start.zig:231:37
```
Best option is to make the allocator nullable and doing an `orelse` on it.

--- a/README.md
+++ b/README.md
@@ -37,6 +37,9 @@ const debug = std.debug;
 const io = std.io;
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     // First we specify what parameters our program can take.
     // We can use `parseParamsComptime` to parse a string into an array of `Param(Help)`
     const params = comptime clap.parseParamsComptime(
@@ -53,6 +56,7 @@ pub fn main() !void {
     var diag = clap.Diagnostic{};
     var res = clap.parse(clap.Help, &params, clap.parsers.default, .{
         .diagnostic = &diag,
+        .allocator = gpa.allocator(),
     }) catch |err| {
         // Report useful error and exit
         diag.report(io.getStdErr().writer(), err) catch {};
@@ -92,6 +96,9 @@ const io = std.io;
 const process = std.process;
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     // First we specify what parameters our program can take.
     // We can use `parseParamsComptime` to parse a string into an array of `Param(Help)`
     const params = comptime clap.parseParamsComptime(
@@ -116,6 +123,7 @@ pub fn main() !void {
     var diag = clap.Diagnostic{};
     var res = clap.parse(clap.Help, &params, parsers, .{
         .diagnostic = &diag,
+        .allocator = gpa.allocator(),
     }) catch |err| {
         diag.report(io.getStdErr().writer(), err) catch {};
         return err;
@@ -219,13 +227,18 @@ const clap = @import("clap");
 const std = @import("std");
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     const params = comptime clap.parseParamsComptime(
         \\-h, --help     Display this help and exit.
         \\-v, --version  Output version information and exit.
         \\
     );
 
-    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{});
+    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{
+        .allocator = gpa.allocator(),
+    });
     defer res.deinit();
 
     // `clap.help` is a function that can print a simple help message. It can print any `Param`
@@ -257,6 +270,9 @@ const clap = @import("clap");
 const std = @import("std");
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     const params = comptime clap.parseParamsComptime(
         \\-h, --help         Display this help and exit.
         \\-v, --version      Output version information and exit.
@@ -264,7 +280,9 @@ pub fn main() !void {
         \\
     );
 
-    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{});
+    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{
+        .allocator = gpa.allocator(),
+    });
     defer res.deinit();
 
     // `clap.usage` is a function that can print a simple help message. It can print any `Param`

--- a/clap.zig
+++ b/clap.zig
@@ -641,12 +641,7 @@ test "Diagnostic.report" {
 
 /// Options that can be set to customize the behavior of parsing.
 pub const ParseOptions = struct {
-    /// The allocator used for all memory allocations. Defaults to the `heap.page_allocator`.
-    /// Note: You should probably override this allocator if you are calling `parseEx`. Unlike
-    ///       `parse`, `parseEx` does not wrap the allocator so the heap allocator can be
-    ///       quite expensive. (TODO: Can we pick a better default? For `parse`, this allocator
-    ///       is fine, as it wraps it in an arena)
-    allocator: mem.Allocator = heap.page_allocator,
+    allocator: mem.Allocator,
     diagnostic: ?*Diagnostic = null,
 };
 

--- a/example/help.zig
+++ b/example/help.zig
@@ -2,13 +2,18 @@ const clap = @import("clap");
 const std = @import("std");
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     const params = comptime clap.parseParamsComptime(
         \\-h, --help     Display this help and exit.
         \\-v, --version  Output version information and exit.
         \\
     );
 
-    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{});
+    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{
+        .allocator = gpa.allocator(),
+    });
     defer res.deinit();
 
     // `clap.help` is a function that can print a simple help message. It can print any `Param`

--- a/example/simple-ex.zig
+++ b/example/simple-ex.zig
@@ -6,6 +6,9 @@ const io = std.io;
 const process = std.process;
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     // First we specify what parameters our program can take.
     // We can use `parseParamsComptime` to parse a string into an array of `Param(Help)`
     const params = comptime clap.parseParamsComptime(
@@ -30,6 +33,7 @@ pub fn main() !void {
     var diag = clap.Diagnostic{};
     var res = clap.parse(clap.Help, &params, parsers, .{
         .diagnostic = &diag,
+        .allocator = gpa.allocator(),
     }) catch |err| {
         diag.report(io.getStdErr().writer(), err) catch {};
         return err;

--- a/example/simple.zig
+++ b/example/simple.zig
@@ -5,6 +5,9 @@ const debug = std.debug;
 const io = std.io;
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     // First we specify what parameters our program can take.
     // We can use `parseParamsComptime` to parse a string into an array of `Param(Help)`
     const params = comptime clap.parseParamsComptime(
@@ -21,6 +24,7 @@ pub fn main() !void {
     var diag = clap.Diagnostic{};
     var res = clap.parse(clap.Help, &params, clap.parsers.default, .{
         .diagnostic = &diag,
+        .allocator = gpa.allocator(),
     }) catch |err| {
         // Report useful error and exit
         diag.report(io.getStdErr().writer(), err) catch {};

--- a/example/usage.zig
+++ b/example/usage.zig
@@ -2,6 +2,9 @@ const clap = @import("clap");
 const std = @import("std");
 
 pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+
     const params = comptime clap.parseParamsComptime(
         \\-h, --help         Display this help and exit.
         \\-v, --version      Output version information and exit.
@@ -9,7 +12,9 @@ pub fn main() !void {
         \\
     );
 
-    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{});
+    var res = try clap.parse(clap.Help, &params, clap.parsers.default, .{
+        .allocator = gpa.allocator(),
+    });
     defer res.deinit();
 
     // `clap.usage` is a function that can print a simple help message. It can print any `Param`

